### PR TITLE
[SPARK-7254][MLlib] Run PowerIterationClustering directly on graph

### PR DESCRIPTION
**Pull request by viirya**
JIRA: https://issues.apache.org/jira/browse/SPARK-7254

### CODE REVIEW

**Comment by viirya** [comment]
Ping @mengxr @jkbradley. Please take a look.

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/PowerIterationClustering.scala [hunk 1]
```diff
@@ -152,7 +152,28 @@ class PowerIterationClustering private[clustering] (
     }
     this
   }
-
+ 
+  /**
+   * Run the PIC algorithm on Graph.
+   *
+   * @param graph an affinity matrix represented as graph, which is the matrix A in the PIC paper.
+   *                     The similarity s,,ij,, represented as the edge between vertices (i, j) must
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/PowerIterationClustering.scala]: fix indentation

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/PowerIterationClustering.scala [hunk 2]
```diff
@@ -213,6 +234,29 @@ object PowerIterationClustering extends Logging {
...
+  def normalize(graph: Graph[Double, Double]): Graph[Double, Double] = {
+    val vD = graph.aggregateMessages[Double](
+      sendMsg = ctx => {
+        val i = ctx.srcId
+        val j = ctx.dstId
+        val s = ctx.attr
+        if (s < 0.0) {
+          throw new SparkException("Similarity must be nonnegative but found s($i, $j) = $s.")
+        }
+        ctx.sendToSrc(s)
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/PowerIterationClustering.scala]: Add `if s > 0.0`?

Comment (viirya) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/PowerIterationClustering.scala]: I think `s` here should be more than 0. So we just need to check `if s < 0.0`?

Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/PowerIterationClustering.scala]: We allow explicit zero similarities from input. But zero edges could be filtered out at this stage.

**Comment by mengxr** [comment]
LGTM. Merged into master. Thanks!

**Pull request closed without merging**